src/v/cloud_storage/topic_manifest.cc

```diff
@@ -173,6 +173,13 @@ struct topic_manifest_handler
 
     bool Null() {
         if (_state == state::expect_value) {
+            if (_key == "retention_bytes") {
```

Comment:
Offline I think you mentioned that we sometimes encode infinite retention with a -1. Should this block at L123 be setting the disabled tristate too?

```
            } else if (_key == "retention_duration") {
                _properties.retention_duration
                  = tristate<std::chrono::milliseconds>(
                    std::chrono::milliseconds(i));
            } else {
```

Reply:
https://github.com/redpanda-data/redpanda/commit/d8b22a2c46d988d1e84d5e3562f912decc19a78b yes but i thought to leave it out from this pr since we are at this stage of the release (but i'm happy to bring it back)

from what i see it's a non issue in the normal use case:
we disable time-based retention by setting it to -1 via the API. internally, this get's translated to a disabled tristate. 
the disabled tristate will be serialized as *nothing* in topic_manifest.json, and the current code will handle it well.

my concern is if we do some manual editing of a topic_manifest.json and try to write "retention.ms" : -1 to disable it. at that point it will be deserialized as a -1ms. but it's a remote possibility

Reply:
I see, thanks for the explanation! I think we should still do it, but a separate PR is fine to get this one out the door

I don't think your "manual edit" scenario is too outlandish: I can see it being done in the heat of an incident, and it only making things worse.